web_widget_bpmn/static/src/js/web_widget_bpmn.js

```diff
@@ -0,0 +1,62 @@
+//Copyright 2019 Therp BV <https://therp.nl>
+//License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+odoo.define('web_widget_bpmn', function(require) {
+    var core = require('web.core');
+    var common = require('web.form_common');
+    var _t = core._t;
+
+    var FieldBPMN = common.AbstractField.extend(
+            common.ReinitializeFieldMixin, {
+
+        
+        build_widget: function(options) {
+            return new BpmnJS(options)
+        },
+
+        initialize_content: function() {
+            var self = this;
+            options = {container: this.$el}
+            this.bpmnjs = this.build_widget(options);
+            if (this.get("effective_readonly")) {
+                // we are readonly therefore remove any listeners so that the
+                // user will not be able to interact with the widget
+                Object.getOwnPropertyNames(
+                    this.bpmnjs.get('eventBus')._listeners).forEach(function(
+                        entry) {
+                        if (entry.match(/^element\..*$/)) {
+                            delete self.bpmnjs.get('eventBus')._listeners[
+                                entry]
+                        }
+                    })}
+            this.bpmnjs.importXML(this.view.datarecord[this.name]);
+            this.bpmnjs.get('canvas').zoom(0.5, 'auto');
```

Comment:
It would be great if this zoom ratio can be configurable or we can pass as an option while defining this widget.